### PR TITLE
[BEAM-4021] Work around tox bug.

### PR DESCRIPTION
**Pull request by udim**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [ ] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [ ] Write a pull request description that is detailed enough to understand:
   - [ ] What the pull request does
   - [ ] Why it does it
   - [ ] How it does it
   - [ ] Why this approach
 - [ ] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

### CODE REVIEW

**Comment by udim** [comment]
R: @chamikaramj , @lukecwik

File: sdks/python/build.gradle [hunk 1]
```diff
@@ -37,7 +38,10 @@ task setupVirtualenv {
       args '-c', ". ${envdir}/bin/activate && pip install --upgrade tox"
     }
   }
-  outputs.files("${envdir}/bin/tox")
+  // Gradle will delete outputs whenever it thinks they are stale. Putting a
```
Comment (aaltay) [inline sdks/python/build.gradle]: Is this a warning to reader? Or is there more work that can be done?

File: sdks/python/tox.ini [hunk 2]
```diff
@@ -33,6 +33,16 @@ extras = test
...
   time
+deps =
+  grpcio-tools==1.3.5
+  cython: cython==0.26.1
+
+# These 2 magic command overrides are required for Jenkins builds.
+# Otherwise we get "OSError: [Errno 2] No such file or directory" errors.
+# Source:
+# https://github.com/tox-dev/tox/issues/123#issuecomment-284714629
+install_command = {envbindir}/python {envbindir}/pip install {opts} {packages}
```
Comment (aaltay) [inline sdks/python/tox.ini]: Do you need the `{envbindir}/python` part? `pip` might already be an executable.

File: sdks/python/tox.ini [hunk 3]
```diff
@@ -33,6 +33,16 @@ extras = test
 whitelist_externals =
   find
   time
+deps =
+  grpcio-tools==1.3.5
+  cython: cython==0.26.1
+
+# These 2 magic command overrides are required for Jenkins builds.
```
Comment (aaltay) [inline sdks/python/tox.ini]: This is bizarre and a good find. I wonder if it is any better with the new tox 3 (released on 4/2).

File: sdks/python/tox.ini (hunk 2 above)
Comment (udim) [reply sdks/python/tox.ini]: Yes, I tried without it and it fails.

File: sdks/python/build.gradle (hunk 1 above)
Comment (udim) [reply sdks/python/build.gradle]: I intended it as documentation about why we're using outputs.dirs vs outputs.files.

File: sdks/python/tox.ini (hunk 3 above)
Comment (udim) [reply sdks/python/tox.ini]: I believe this is already running using tox 3.0.0.

**Comment by aaltay** [comment]
Thanks for answering my questions. It looks good other than the test failure.

**Comment by udim** [comment]
I don't know why the maven build fails to run `pip --version` while the gradle one succeeds (same machine).

**Pull request merged**